probe-rs/src/core/registers.rs

```diff
@@ -293,151 +351,220 @@ impl<T> RegisterValueResultExt<T> for Result<T, Infallible> {
     }
 }
 
-/// Register description for a core.
+/// A static array of all the registers ([`CoreRegister`]).
 #[derive(Debug, PartialEq)]
-pub struct RegisterFile {
-    pub(crate) platform_registers: &'static [RegisterDescription],
-
-    /// Register description for the program counter
-    pub(crate) program_counter: &'static RegisterDescription,
-
-    pub(crate) stack_pointer: &'static RegisterDescription,
-
-    pub(crate) return_address: &'static RegisterDescription,
-
-    pub(crate) frame_pointer: &'static RegisterDescription,
+pub struct RegisterFile(Vec<CoreRegister>);
 
-    pub(crate) argument_registers: &'static [RegisterDescription],
-
-    pub(crate) result_registers: &'static [RegisterDescription],
-
-    pub(crate) msp: Option<&'static RegisterDescription>,
-
-    pub(crate) psp: Option<&'static RegisterDescription>,
-
-    pub(crate) psr: Option<&'static RegisterDescription>,
-
-    pub(crate) fp_status: Option<&'static RegisterDescription>,
-
-    pub(crate) fp_registers: Option<&'static [RegisterDescription]>,
+impl RegisterFile {
+    /// Construct a new register file from a vector of [`CoreRegister`]s, and panics if it does not contain at least the essential entries.
+    pub fn new(core_registers: Vec<CoreRegister>) -> RegisterFile {
+        let register_file = RegisterFile(core_registers);
+        // Ensure we have the minimum required registers.
+        let _ = register_file.program_counter().unwrap();
```

Comment:
If this is required, I would prefer it as a separate argument, and not as something which can fail at runtime.

Reply:
@Tiwalun Because of the way the `RegisterFile` contents is created, it doesn't really make sense for me to pass those as parameters (unless I'm missing something obvious). 

To ensure this doesn't fail at runtime, I've added tests for all the currently supported `RegisterFile` statics to ensure they have these required register entries.

Reply:
Don't all cores have those registers? so we could make those statically accessible?

Reply:
That is a good point/suggestion. Tx :)